src/main/java/com/breskul/bibernate/persistence/GenericDao.java

```diff
@@ -301,7 +311,11 @@ public <T> int executeUpdate(EntityKey<T> entityKey, Object... parameters) {
     }
   }
 
-  public <T> String prepareUpdateQuery(EntityKey<T> entityKey) {
+  private <T> String prepareUpdateQuery(EntityKey<T> entityKey) {
+    if (EntityUtil.isDynamicUpdate(entityKey.entityClass())) {
+      return prepareDynamicUpdateQuery(entityKey);
```

Comment:
If you decide to put a dynamic update into another method, then I advise putting the non-dynamic code below into a separate method too, thus here you will need to use the ternary operator to select what method to execute.

Reply:
good point, done